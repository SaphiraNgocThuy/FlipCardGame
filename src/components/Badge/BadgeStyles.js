import styled from "styled-components/native";
import { padding, fontSize, borderRadius } from "../../utils/commonStyles";
import { theme } from "../../utils/theme";

export const BadgeContainer = styled.View`
  padding: ${padding.md}px ${padding.lg}px;
  margin-right: ${padding.md}px;
  border-radius: ${borderRadius}px;
  border: 2px solid ${theme.darkGrey};
  justify-content: center;
`;

export const BadgeText = styled.Text`
  font-size: ${fontSize.lg}px;
  font-weight: bold;
  color: ${theme.darkGrey};
`;
