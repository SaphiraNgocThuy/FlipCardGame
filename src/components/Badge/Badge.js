import React, { memo } from "react";
import PropTypes from "prop-types";
import { BadgeText, BadgeContainer } from "./BadgeStyles";

const CurvedBadge = ({ steps }) => {
  return (
    <BadgeContainer>
      <BadgeText>STEPS: {steps}</BadgeText>
    </BadgeContainer>
  );
};

CurvedBadge.propTypes = {
  steps: PropTypes.number.isRequired,
};

export default memo(CurvedBadge);
