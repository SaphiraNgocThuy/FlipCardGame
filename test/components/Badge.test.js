import React from 'react';
import renderer from 'react-test-renderer';
import Badge from '../../src/components/Badge/Badge';

it('renders correctly', () => {
  const tree = renderer.create(<Badge steps={4} />).toJSON();
  expect(tree).toMatchSnapshot();
});
