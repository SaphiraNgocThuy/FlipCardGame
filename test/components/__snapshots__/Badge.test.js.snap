// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#424c58",
        "borderRadius": 30,
        "borderStyle": "solid",
        "borderWidth": 2,
        "justifyContent": "center",
        "marginRight": 16,
        "paddingBottom": 16,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#424c58",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    STEPS: 
    4
  </Text>
</View>
`;
